File: app/src/main/java/com/example/ruburger/model/OrderManager.java
package com.example.ruburger.model;

import java.util.ArrayList;
import java.util.List;

/**
 * Singleton for sharing the current order between Activities.
 */
public class OrderManager {
    private static final OrderManager INSTANCE = new OrderManager();
    private final List<MenuItem> currentOrder = new ArrayList<>();

    /** Private constructor prevents external instantiation */
    private OrderManager() { }

    /** Returns the single shared instance */
    public static OrderManager getInstance() {
        return INSTANCE;
    }

    /** Adds an item to the current order */
    public void addItem(MenuItem item) {
        currentOrder.add(item);
    }

    /** Returns the list of items in the current order */
    public List<MenuItem> getOrder() {
        return currentOrder;
    }

    /** Clears the current order */
    public void clear() {
        currentOrder.clear();
    }
}

File: app/src/main/java/com/example/ruburger/OrderManager.kt
package com.example.ruburger.singleton

import com.example.ruburger.model.MenuItem

class OrderManager private constructor() {

    private val currentOrderItems = mutableListOf<MenuItem>()

    companion object {
        private var instance: OrderManager? = null

        fun getInstance(): OrderManager {
            if (instance == null) {
                instance = OrderManager()
            }
            return instance!!
        }
    }

    fun addItem(item: MenuItem) {
        currentOrderItems.add(item)
    }

    fun getItems(): List<MenuItem> {
        return currentOrderItems
    }

    fun clear() {
        currentOrderItems.clear()
    }
}
